src/validation.rs

```diff
@@ -417,4 +445,27 @@ impl Validator {
 
         self.stmt_validator.validate_statement(statement, context);
     }
+
+    fn validate_assignment(
+        &mut self,
+        left_type: &typesystem::DataType,
+        right_type: &typesystem::DataType,
+        location: SourceRange,
+        index: &Index,
+    ) {
+        let left_type_info = index.find_intrinsic_type(left_type.get_type_information());
```

Comment:
ok this is where we need the new find_intrinsic_behavior. I suggest this:

- rename this method to: validate_call_parameter_assignment since normal assignments `a := 3` will not be validated by this method
- for the left we pass the VariableIndexEntry here, so we can even see if this is a VAR_IN_OUT
- if it is a VAR_IN_OUT we do the Pointer-behavior in here instead of in the find_intrinsic_type method / alternativeley i believe you could do this as well with the DataType, but then you need to check if it is a AutoDeref-Pointer